### PR TITLE
fix(manager/sbt): allow star comment

### PR DESCRIPTION
**Pull request by cjtim**
## Changes

* allow comment `/** comment */`

## Context
* problem
* Cannot having comment after dependency line `libraryDependencies += "org.example" %% "bar" % "0.0.2" /** comment **/`
* The replace function gets everything after `=`
* So it unable to parse that dependency

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/sbt/extract.ts [hunk 1]
```diff
@@ -12,7 +12,9 @@ import type { PackageDependency, PackageFile } from '../types';
 import type { ParseContext, ParseOptions } from './types';
 
 const stripComment = (str: string): string =>
-  str.replace(regEx(/(^|\s+)\/\/.*$/), '');
+  str
+    .replace(regEx(/(^|\s+)\/\/.*$/), '')
+    .replace(regEx(/\/(\*)+.*(\*)+\//), ''); /** allow comment */
```
Comment (viceice) [inline lib/modules/manager/sbt/extract.ts:17]: i think this will be to greedy.

```
/** t */
value
/** c */
```

would be completely removed

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/modules/manager/sbt/extract.ts (hunk 1 above)
Comment (cjtim) [reply lib/modules/manager/sbt/extract.ts:17]: This invoke be `extractPackageFile` in line 355
It actually run 1 line.

**Comment by zharinov** [comment]
I think actually we can benefit from using parser library here, rather then bringing more and more regex mess.

**Comment by zharinov** [comment]
I am okay with this PR though, but please add relevant test cases for this

File: lib/modules/manager/sbt/extract.ts (hunk 1 above)
Comment (zharinov) [inline lib/modules/manager/sbt/extract.ts:17]:
```suggestion
    .replace(regEx(/(?:^|\s+)\/\/.*$/), '')
    .replace(regEx(/\/\*.*?\*\//g), '');
```

**Review by zharinov — CHANGES_REQUESTED** [review]

**Comment by cjtim** [comment]
@zharinov 
> I am okay with this PR though, but please add relevant test cases for this

added

**Review by viceice — DISMISSED** [review]

**Review by viceice — COMMENTED** [review]
needs prettify

**Review by rarkins — APPROVED** [review]

**Pull request merged**